tools/db/deleteLogsFromActivations.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2015-2016 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import time
+import couchdb.client
+
+try:
+    long        # Python 2
+except NameError:
+    long = int  # Python 3
+
+DAY = 1000 * 60 * 60 * 24
+
+#
+# Delete activations
+#
+def deleteLogsFromOldActivations(args):
+    db = couchdb.client.Server(args.dbUrl)[args.dbName]
+    endkey = long(time.time() * 1000) - args.days * DAY
+    while True:
+        activations = db.view("logCleanup/byDateWithLogs", limit=args.docsPerRequest, start_key=0, end_key=endkey, include_docs=True)
+        if activations:
+            def transform(viewResult):
+                doc = viewResult.doc
+                doc["logs"] = []
+                return doc
+
+            activationsWithoutLogs = map(lambda activation: transform(activation), activations)
```

Comment:
The intermediate lambda is not necessary in the approach you take: `map(transform, activations)` should work here.